gcloud/_helpers.py

```diff
@@ -302,14 +303,30 @@ def _total_seconds(offset):
 def _rfc3339_to_datetime(dt_str):
     """Convert a string to a native timestamp.
 
+    .. note::
+
+       Python datetimes do not support nanosecond precision;  this function
+       therefore truncates such values to microseconds.
+
     :type dt_str: str
     :param dt_str: The string to convert.
 
     :rtype: :class:`datetime.datetime`
     :returns: The datetime object created from the string.
     """
-    return datetime.datetime.strptime(
-        dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    try:
+        return datetime.datetime.strptime(
+            dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    except ValueError:  # maybe nanosecond resoultion
+        prefix, suffix = dt_str.split('.')
+        fraction, zulu = suffix[:-1], suffix[-1]
```

Comment:
Both `dt_str.split('.')` and `suffix[-1]` will lead to strange stacktraces if the input is not nanosecond (e.g. `dt_str = 'abc'` without a `.` or `dt_str = 'abc.d.e'` with too many `.` or `dt_str = 'abc.'` with an empty string for the suffix)